### ISSUE TITLE
One of the major issue which need to be fixed ASAP

### ISSUE THREAD

**Issue by rocky8roy** [wlanjie/trinity#107]
Hi,
thumbnail drag issue is there in old and new library both.
Anyone reading this may already have faced this issue and all are struggling to solve this as this is not device specific or anything.
we know there is also memory issue which will require time to fix it but this thumbnail drag issue can be fixed by owner of this wonderful library. 
we know you are very busy but kindly look into this and try to fix it asap.

**Comment by WU1208** [comment]
Have you used version 0.3.0?

**Comment by badcodercpp** [comment]
@WU1208 is it fixed in 0.3.0 ?? on which device you are running ?? it's still crash on xiaomi note-4,5 (arm64) , "av_frame_unref" is causing null dereference on some devices.

**Comment by WU1208** [comment]
@badcodercpp Running in Redmi Note 8, Sometimes it turns black, but it doesn't crash. Now I don't know how to use Updateeffect,it does not work in old demo

**Comment by WU1208** [comment]
@badcodercpp ![image](https://user-images.githubusercontent.com/13074367/95822299-390a5f00-0d5e-11eb-89c1-ed5777e198a7.png)

**Comment by rocky8roy** [comment]
Has anyone solved the thumbnail issue?

**Comment by badcodercpp** [comment]
yeah find av_frame_unref(frame) in queue.c, check if frame is available in FramePool or not before calling "av_frame_unref(frame)", Actually problem is that on video seek library is calling AVFrameFlush to flush the frame pool and free frame from buffer but on fast drag flush free the buffer first and then call of "av_frame_unref(frame)" fails as frame is NULL that time, add null check on frame or find frame in pool and if find is successful then call "av_frame_unref(frame)" or simply remove this "av_frame_unref(frame)" line , as this will be taken care by the FFMPEG library itself, 
NOTE: - DO NOT COPY frame if you are removing "av_frame_unref(frame)" line or else it will crash on memcpy.

**Comment by rocky8roy** [comment]
Thanks.. Will try this.
Is there any sample code you have implemented which is working without any
crashes?

**Comment by badcodercpp** [comment]
yeah if you are not copying frame then replace implementation of "frame_pool_unref_frame" in queue.c with following : 

void frame_pool_unref_frame(FramePool *pool, AVFrame *frame){
    //av_frame_unref(frame); // remove this line completely memory free will be handled by libavutil
    pool->count--;
}

**Comment by WU1208** [comment]
@badcodercpp hey,you used version 0.3.0?Or 0.9.1?

**Comment by badcodercpp** [comment]
@WU1208 i used recent one it's 0.3.0 i guess, sure i will help you , but having very schedule in day time , will take a look in late evening , meanwhile drop me mail with UpdateEffect signature what you are using on badcodercpp@gmail.com. i will  look into that in late evening and revert.

**Comment by WU1208** [comment]
@badcodercpp Thank you, 0.3.1 fixed this problem, UpdateEffect

**Comment by rocky8roy** [comment]
@WU1208 @badcodercpp is there any sample working code
Possible to share your sample code which is not crashing automatically?
for me version 0.3.2 also not working properly
if anyone of you have successfully implemented the code in any of the version then kindly share it
you can also email the code on rocky8roy@gmail.com
thanks
